core/templates/pages/admin-page/roles-tab/topic-manager-role-editor-modal.component.ts

```diff
@@ -36,34 +38,47 @@ export class TopicManagerRoleEditorModalComponent implements OnInit {
 
   constructor(
     private activeModal: NgbActiveModal,
-    private adminBackendApiService: AdminBackendApiService
+    private adminBackendApiService: AdminBackendApiService,
+    private alertsService: AlertsService
   ) {}
 
   private updateTopicIdsForSelection(): void {
     this.topicIdsForSelection = Object.keys(this.topicIdToName).filter(
       topicId => !this.managerInTopicsWithId.includes(topicId));
+    this.newTopicId = this.topicIdsForSelection[0];
   }
 
   addTopic(): void {
     this.managerInTopicsWithId.push(this.newTopicId);
     this.topicIdInUpdate = this.newTopicId;
     this.newTopicId = null;
-    this.adminBackendApiService.addUserRoleAsync(
-      'TOPIC_MANAGER', this.username, this.topicIdInUpdate).then(()=> {
-      this.newTopicId = null;
+    this.adminBackendApiService.assignManagerToTopicAsync(
+      this.username, this.topicIdInUpdate).then(()=> {
       this.topicIdInUpdate = null;
       this.updateTopicIdsForSelection();
+    }, data => {
+      let topicIdIndex = this.managerInTopicsWithId.indexOf(this.newTopicId);
+      this.managerInTopicsWithId.splice(topicIdIndex, 1);
```

Comment:
managedTopicIds perhaps? I find managerInTopicsWithId hard to parse.

Reply:
Done!